modules/pico_fragments.c

```diff
@@ -450,12 +450,12 @@ void pico_ipv6_process_frag(struct pico_ipv6_exthdr *frag, struct pico_frame *f,
                 return;
             }
 
-        if (pico_tree_insert(&ipv6_fragments, temp)) {
-            frag_dbg("Could not insert temp in tree\n");
-            PICO_FREE(temp);
-            return;
+            if (pico_tree_insert(&ipv6_fragments, temp)) {
```

Comment:
Could you rework these similar blocks into functions? Will do good for the TICS score